signers/signer-evm/src/signer.ts

```diff
@@ -52,12 +49,15 @@ const waitWithMempoolCheck = async (
 };
 
 export class DefaultEvmSigner implements GenericSigner<EvmTransaction> {
-  private signer: providers.JsonRpcSigner;
-  private provider: providers.Web3Provider;
+  private provider: BrowserProvider;
 
   constructor(provider: ProviderType) {
-    this.provider = new providers.Web3Provider(provider);
-    this.signer = this.provider.getSigner();
+    /*
+     * Connect to the MetaMask EIP-1193 object. This is a standard
+     * protocol that allows Ethers access to make all read-only
+     * requests through MetaMask.
+     */
```

Comment:
I recommend either removing this comment or shortening it.

```suggestion
    // get read-only access to the provider
```

Reply:
Improved comment